public/components/directives/ui-edit-rights.js

```diff
@@ -73,10 +73,32 @@ export const uiEditRights = (wfComposerService) => ({
       })
     };
 
+    /**
+     * Because we go back to Composer to set this information, and then wait for Composer
+     * to report the new value via flexible-content-stream, sometimes workflow-frontend will
+     * update before Composer has a chance to report the new value.
+     *
+     * Our client cannot distinguish between these sorts of updates, so we lock these values
+     * once they're changed for a short duration after editing.
+     */
+    const lockRightsValuesForDuration = (shouldHaveRights) => {
+      const cancelWatch = scope.$watch("contentItem", contentItem => {
+        contentItem.item.rightsSyndicationAggregate = shouldHaveRights;
+        contentItem.item.rightsSubscriptionDatabases = shouldHaveRights;
+        contentItem.item.rightsDeveloperCommunity = shouldHaveRights;
+      });
+
+      // Lock rights data for one polling cycle.
+      setTimeout(cancelWatch, 5000);
```

Comment:
What would happen if user tried to change the value twice in quick succession? Would the first update work but the second one not? And if latter, then would we want to grey out the boxes in the UI until they are available again, or display some kind of loading indicator?

Reply:
It shouldn't be possible to change the value twice on a single piece of content with 'add all rights', as the button disappears as soon as it's used. ('Remove all rights' is no longer a thing 😢) 

There will be an odd synchronisation issue if the user _then_ opens the drawer and observes the behaviour of the syndication checkboxes, which will sometimes regress on the first update, before likely syncing up on the second.

To improve this, we could simultaneously write to Composer, and Workflow backend locally – workflow backend then anticipates the change, and does not update the client with incorrect information, but will update via the stream shortly.

Reply:
(Alongside this, trying to figure out how much effort to expend given that the service is eventually consistent and we know the workflow may be somewhat temporary!)